File: AtlasWeb/Core/Modules/Tibbi_Surec/Sarf_Malzeme_Modulu/TreatmentMaterialFormViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using TTObjectClasses;

namespace Core.Models
{
    public class TreatmentMaterialFormViewModel
    {
        public object _selectedMaterialValue
        {
            get;
            set;
        }

        public TTObjectClasses.Store _selectedStoreValue
        {
            get;
            set;
        }

        //public TTObjectClasses.BaseTreatmentMaterial AddedTreatmentMaterial { get; set; }
        //public TTObjectClasses.Material material { get; set; }
        //public string Barcode { get; set; }
        //public TTObjectClasses.StockCard StockCard { get;set;}
        //public TTObjectClasses.DistributionTypeDefinition DistributionTypeDef { get; set; }
        //public string DistributionType { get; set; }
        public List<AddedTreatmentMaterialsViewModel> AddedTreatmentMaterials
        {
            get;
            set;
        }

        public DateTime ActionDate
        {
            get;
            set;
        }

        public int countForDailyOperations {get; set;}

        public string ProtocolNo { get; set; }
    }

    public class GetTreatmentMaterialDetail_Input
    {
        public TTObjectClasses.BaseTreatmentMaterial Material { get;set;}
    }

public class AddedTreatmentMaterialsViewModel
{
    public TTObjectClasses.BaseTreatmentMaterial AddedTreatmentMaterial
    {
        get;
        set;
    }

    public TTObjectClasses.Material material
    {
        get;
        set;
    }

    public string Barcode
    {
        get;
        set;
    }
        
        public TTObjectClasses.StockCard StockCard
    {
        get;
        set;
    }

    public TTObjectClasses.DistributionTypeDefinition DistributionTypeDef
    {
        get;
        set;
    }

    public string DistributionType
    {
        get;
        set;
    }
        public string drugSpecification
        {
            get;
            set;
        }
        public DrugReportType DrugReportType
    {
        get;
        set;
    }

        public List<MaterialProcedureViewModel> MaterialProcedureViewModelList
        {
            get;
            set;
        }


        public SubEpisode subEpisode { get; set; } //Gridde protokol no'yu gosterebilmek icin
        public EpisodeAction episodeAction { get; set; } //Gridde action type'ı gosterebilmek icin
}

    public class AddedTreatmentMaterialInputDVO
    {
        public List<TreatmentMaterialInputDVODetail> AddedTreatmentMaterials
        {
            get;
            set;
        }

        public string EpisodeActionObjectDefID
        {
            get;
            set;
        }

        public DateTime ActionDate
        {
            get;
            set;
        }

        public Guid SubEpisodeGuid
        {
            get;
            set;
        }

        public string TreatmentMaterialTypeName
        {
            get;
            set;
        }
        public Guid EpisodeActionMasterResourceId
        {
            get;
            set;
        }

    }

public class TreatmentMaterialInputDVODetail
{
    public string MaterialObjectID
    {
        get;
        set;
    }
    public double Amount
    {
        get;
        set;
    }
    public DateTime? TransactionDate
    {
        get;
        set;
    }
}

public class TreatmentMaterialStartUpViewModel
{
    public DateTime? MaxDate
    {
        get;
        set;
    }
    public DateTime? MinDate
    {
        get;
        set;
    }
    public DateTime DefaultDate
    {
        get;
        set;
    }

    public string ProtocolNo 
        {
            get;
            set;
        }


}
    public class MaterialProcedureViewModel
    {
        public string ProcedureName;
        public System.Guid ProcedureObjectId;
        public System.Guid? ProcedureRequestFormDetailObjectId;
        public bool IsAdditionalApplication;
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Sarf_Malzeme_Modulu/TreatmentMaterialFormViewModel.ts

import { BaseTreatmentMaterial, DrugReportType, SubEpisode, EpisodeAction } from 'NebulaClient/Model/AtlasClientModel';
import { Material } from 'NebulaClient/Model/AtlasClientModel';
import { StockCard } from 'NebulaClient/Model/AtlasClientModel';
import { DistributionTypeDefinition } from 'NebulaClient/Model/AtlasClientModel';
import { Store } from 'NebulaClient/Model/AtlasClientModel';
import { Type } from 'NebulaClient/ClassTransformer';
import { BaseViewModel } from 'NebulaClient/Model/BaseViewModel';
import { Guid } from 'NebulaClient/Mscorlib/Guid';

export class TreatmentMaterialFormViewModel extends BaseViewModel {
    @Type(() => Object)
    public _selectedMaterialValue: Object;
    @Type(() => Store)
    public _selectedStoreValue: Store;
    //public AddedTreatmentMaterial: BaseTreatmentMaterial;
    //public material: Material;
    //public Barcode: string;
    //public StockCard: StockCard;
    //public DistributionTypeDef: DistributionTypeDefinition;
    //public DistributionType: string;
    @Type(() => AddedTreatmentMaterialsViewModel)
    public AddedTreatmentMaterials: Array<AddedTreatmentMaterialsViewModel>;
    @Type(() => Date)
    public ActionDate: Date;
    countForDailyOperations: number = 0;
}

export class AddedTreatmentMaterialsViewModel {
    @Type(() => BaseTreatmentMaterial)
    public AddedTreatmentMaterial: BaseTreatmentMaterial;
    @Type(() => Material)
    public material: Material;
    public Barcode: string;
    @Type(() => StockCard)
    public StockCard: StockCard;
    @Type(() => DistributionTypeDefinition)
    public DistributionTypeDef: DistributionTypeDefinition;
    public DistributionType: string;
    @Type(() => Date)
    public TransactionDate: Date;
    public drugSpecification: string;
   public DrugReportType: DrugReportType;
    @Type(() => MaterialProcedureViewModel)
    MaterialProcedureViewModelList: Array<MaterialProcedureViewModel> = new Array<MaterialProcedureViewModel>();
    @Type(() => SubEpisode)
    public subEpisode: SubEpisode;
    @Type(() => EpisodeAction)
    public episodeAction: EpisodeAction;
}



export class AddedTreatmentMaterialInputDVO {
    public AddedTreatmentMaterials: Array<TreatmentMaterialInputDVODetail>;
    public EpisodeActionObjectDefID: string;
    @Type(() => Date)
    public ActionDate: Date;
    @Type(() => Guid)
    public SubEpisodeGuid: Guid;
    public TreatmentMaterialTypeName: string;
    @Type(() => Guid)
    public EpisodeActionMasterResourceId: Guid;
}

export class TreatmentMaterialInputDVODetail {
    public MaterialObjectID: string;
    public Amount: number;
    @Type(() => Date)
    public TransactionDate: Date;
}

export class TreatmentMaterialStartUpViewModel {
    @Type(() => Date)
    public MaxDate: Date;
    @Type(() => Date)
    public MinDate: Date;
    @Type(() => Date)
    public DefaultDate: Date;
    public ProtocolNo: string;
}

export class BaseTreatmentMaterialUTSUsageNotificationResultViewModel {
    @Type(() => Guid)
    public ObjectId: Guid;
    public UTSUseNotificationState: string;
    public Message: string;
}
export class GetTreatmentMaterialDetail_Input{
    @Type(() => BaseTreatmentMaterial)
    public Material: BaseTreatmentMaterial;
}

export class MaterialProcedureViewModel {
    ProcedureName: string;
    @Type(() => Guid)
    ProcedureObjectId: Guid;
    @Type(() => Guid)
    ProcedureRequestFormDetailObjectId: Guid;
    IsAdditionalApplication: Boolean;
}